compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java

```diff
@@ -175,6 +175,20 @@ boolean isDefaultModule() {
         allModuleLoadRequests = new LinkedHashSet<>();
         Set<ModuleLoadRequest> moduleLoadRequests = new LinkedHashSet<>();
         for (DocumentContext docContext : srcDocContextMap.values()) {
+            for (ModuleLoadRequest request : docContext.moduleLoadRequests(moduleId, PackageDependencyScope.DEFAULT)) {
+                if (allModuleLoadRequests.contains(request) && !request.locations().isEmpty()) {
```

Comment:
Do we need to check for empty locations here?

Reply:
Used it to avoid unnecessary looping since if a location is not there, we can't add locations to `allModuleLoadRequest`